### ISSUE TITLE
Hide/Show annotations?

### ISSUE THREAD

**Issue by Techn1x** [chartjs/chartjs-plugin-annotation#95]
Hi, is there any way to hide & show the individual annotations with a property? (Rather than removing the object from the `annotations` array?)

For example, Chart.JS allows you to define a `display`: true/false/callback on things like the legend

Thanks!

**Comment by corentingi** [comment]
I am looking for the same thing
To actually emulate this, it is needed to delete and recreate the annotations...

**Comment by compwright** [comment]
Not currently. If you'd like to take a crack at it, we'll entertain a PR! Thanks!

**Comment by Techn1x** [comment]
I might take a look at doing a PR at some point. Any guidance?

I've taken a look at the chartjs-plugin-datalabels project which uses a `display` attribute (there's a global one, and a per-data one), all of the logic for it is in this file https://github.com/chartjs/chartjs-plugin-datalabels/blob/a56a4d1bf733976507205046d189d5fd6c4f3005/src/plugin.js

Just having trouble envisioning how I might apply the same to this project

Hopefully when I take a deeper look later it might become more apparent to me

**Comment by yusufozturk** [comment]
@Techn1x  @corentingi 
You can make this by adding a "display" option into your annotation. Or you can hide them entirely, or you can just hide annotations individually.

I'm going to make a pull request for this. But for now, you can make following changes.

First, add a new option called "display" into default options:

![image](https://user-images.githubusercontent.com/928600/34516201-57a5bc3c-f076-11e7-9659-1019b5e57297.png)

Then add "canHide" option to check them individually:

![image](https://user-images.githubusercontent.com/928600/34516241-7f8cfb98-f076-11e7-88a9-a58dc4784740.png)

So if you add "canHide: true" into your annotation and if you leave "display: false", then these annotations will not be visible.

How can you enable/disable visibility of the annotations? Just enable "display" option by setting it to "display: true/false" and then update your chart. Like in example:

`chart.options.annotation.display = true;
chart.update();`

That's it :)

**Comment by yusufozturk** [comment]
Or another alternative:

![image](https://user-images.githubusercontent.com/928600/34517210-660333be-f07a-11e7-9c61-c89f2a0de224.png)

So you can also hide them individually by setting "isHidden:true" option. I used "isHidden" instead of "display" so nobody will have issue after version upgrade. Otherwise everybody has to update their annotation config which is not very pleasant.

**Comment by GevaertB** [comment]
will this be in the next release?

**Comment by Kibo007** [comment]
Is this implemented in plugin? Can I show / hide annotations?

**Comment by Techn1x** [comment]
Inspired by @yusufozturk I've added a PR that adds this feature https://github.com/chartjs/chartjs-plugin-annotation/pull/141

It's a bit more advanced in that it can be dynamically changed updated after initial draw of annotations. It also uses `display` attribute, but enables it by default for all annotations.

Took me a year to get it done but oh well :)

**Comment by matthewhegarty** [comment]
It looks like this got close to being completed, but didn't get over the line.  @Techn1x Any chance it could be completed?

**Comment by poli44** [comment]
function UnDraw() {
            window.myLine.config.data.datasets.forEach(function (dataset) {
                window.myLine.options.annotation.annotations.pop();
            });
            window.myLine.update();
        };
        function Draw(a) {
            window.myLine.options.annotation.annotations.push(
                {
                    type: 'box',
                    drawTime: 'afterDraw',
                    xScaleID: 'x-axis-0',
                    yScaleID: 'y-axis-0',
                    xMax: a * 1000,
                    xMin: a * 1000,
                    yMax: MAXY,
                    yMin: MINY,
                    borderColor: 'rgba(255,102,0,1)',
                    borderWidth: 1,
                    backgroundColor: 'rgba(255,102,0,1)',
                });
            window.myLine.update();
        };

**Comment by ThaDaVos** [comment]
Any ETA on the implementation?

**Comment by jgatto1** [comment]
try to set the value to a value out of the axis:

```
                    lineElement.options.value = -1;
                    chartInstance.update(0);
```

**Comment by joshfleming** [comment]
I was trying to do this in in React. 
I saw that the README mentions a display property, but it doesn't seem to be doing anything at the moment. 
Instead I basically made it invisible with colors. Not a great solution ... just an option if you need it. 

```
annotation: {
      drawTime: 'beforeDatasetsDraw' as const,
      annotations: [{
        id: 'hline1',
        type: 'line' as const,
        mode: 'horizontal' as const,
        scaleID: 'y-axis-0',
        value: valueVar,
        borderColor: showHideVar? "orange": "transparent",
        borderWidth: 2,
        borderDash: [2, 2],
        borderDashOffset: 5,
        label: {
          backgroundColor: showHideVar? "orange": "transparent",
          content: showHideVar? "Text": "",
          enabled: true
        }
      }]
    }
   ```

**Comment by stockiNail** [comment]
@joshfleming seeing the annotation configuration you posted, I have the feeling you are not using the version 1.x (or master) where `display` options has been implemented.
Which plugin version are you using?

**Comment by etimberg** [comment]
Closing since this is implemented in v1.0.0-beta.4 https://jsfiddle.net/892t1r03/
